apis/externalsecrets/v1alpha1/pushsecret_types.go

```diff
@@ -97,6 +97,9 @@ type PushSecretMatch struct {
 type PushSecretData struct {
 	// Match a given Secret Key to be pushed to the provider.
 	Match PushSecretMatch `json:"match"`
+	// Metadata is a map of metadata attached to the secret
+	// +optional
+	Metadata map[string]map[string]string `json:"metadata,omitempty"`
```

Comment:
I first was a bit hesitant to have it as a fixed type of `map[string]map[string]string`, just for the reason of future-proofing: Imagine a provider that allows to store different kind of metadata, e.g. a indefinitely nested map or to attach `[]string` to a secret. That can not be easily achieved with the proposed structure.

As an alternative, we could make it a `interface{}` type, parse it in the provider and return an error when appropriate.

However, looking at the existing providers today, it doesn't look like it is needed today, and, from an API design perspective, i don't see someone implementing a system like that (famous last words :laughing:).

With that being said: I'm fine to leave it as-is. Though i would suggest to add a comment to further explain the structure of this field, e.g.: The structure of `metadata` is provider specific, please look it up in the provider documentation.

Reply:
Hmm, yes I think that is a good point. Since we don't know what kind of metadata we (will) need to support at this point, I believe using `interface{}` makes it much more robust for future changes. Let me change it so 🙂